src/sst/elements/vanadis/inst/regfile.h

```diff
@@ -82,50 +90,63 @@ class VanadisRegisterFile
         assert(reg < count_fp_regs);
         assert((offset + len) <= fp_reg_width);
 
-        uint8_t* reg_ptr = (uint8_t*) &fp_reg_storage[reg * fp_reg_width];
+        int index = get_reg_index(reg,1);
+        uint8_t* reg_ptr = (uint8_t*) &fp_reg_storage[index];
+        // output->verbose(CALL_INFO, 16, 0, "CopyFromFPReg: reg=%d fp_reg_width=%d target_tid=%d WARP_SIZE=%d count_fp_regs=%d index = %d\n",
+        // reg, fp_reg_width,target_tid, WARP_SIZE, count_fp_regs, index);
 
         for(auto i = 0; i < len; ++i) {
             values[i] = reg_ptr[offset + i];
+            // printf("reg[%d][%d]=%d\n",reg, offset + i,reg_ptr[offset+i]);
         }
+        
+        
     }
 
     void copyFromIntRegister(uint16_t reg, uint32_t offset, uint8_t* values, uint32_t len) {
         assert(reg < count_int_regs);
         assert((offset + len) <= int_reg_width);
-
-        uint8_t* reg_ptr = (uint8_t*) &int_reg_storage[reg * int_reg_width];
+        
+        int index = get_reg_index(reg, 0);
+        uint8_t* reg_ptr = (uint8_t*) &int_reg_storage[index];
+        // output->verbose(CALL_INFO, 16, 0, "CopyFromIntReg: reg=%d int_reg_width=%d target_tid=%d WARP_SIZE=%d count_int_regs=%d index= %d len=%d\n",
```

Comment:
More debugging; please check all of the files for code that has been commented-out for deubgging